File: testdata/fixtures/go-service/internal/order/order.go
package order

import "github.com/acme/goservice/internal/pricing"

// Entity is embedded by every persisted type.
type Entity struct {
	ID string
}

func (e Entity) Key() string { return "order:" + e.ID }

type Line struct {
	Price pricing.Money
	Qty   int
}

type Order struct {
	Entity
	Coupon string
	lines  []Line
}

// Service totals orders.
type Service interface {
	Total(o *Order) pricing.Money
}

File: testdata/fixtures/java-service/src/main/java/com/acme/order/Order.java
package com.acme.order;

import java.util.List;
import com.acme.pricing.Money;

public class Order {
    private final String id;
    private final List<Money> lines;

    public Order(String id, List<Money> lines) {
        this.id = id;
        this.lines = lines;
    }

    public String id() {
        return id;
    }

    public List<Money> lines() {
        return lines;
    }
}
